### DISCUSSION TITLE
I THINK that the ESLint Parser's peerDependency on @babel/core @ >= 7.11.0 should be updated to be >= 7.17.0

### DISCUSSION THREAD

**Discussion by matty-at-rdc** [babel/babel#15552] [General] [answered]
Not sure how to do this discussion, but here goes!

On older installs whose lockfile resolutions are **not** invalid based on the current peerDependency on @babel/core @ >= 7.11.0 in the ESLint parser you run the risk of a perfectly "valid" dependency configuration causing failures on rules dealing with TemplateLiterals since at pre-7.17.0 `tl.doubleAt` is `undefined` and the monkey-patched `Template` type in the ESLint Parser's _convertToken.cjs_ is also `undefined` thus causing `Template` to be classified as `Punctuator`. 

Here is a repo demonstrating the issue: https://github.com/matty-at-rdc/literally-what

Updating this peerDependency would allow users using the ESLint Parser to be made aware by way of a warning that they aren't protected from issues since they don't have a valid peerDependency for @babel/core. 

Hope this made sense, and please guide me if I missed something important in the 📖 that explains this choice.

**Comment by JLHwung** [comment, marked as answer]
Thanks for the reproduction repo. This is indeed a compatibility bug. If `token.type.label` is `undefined`, we should classify it as `Template` first, so whether `tl.doubleAt` is defined or not (depends on the babel/core version), the parser should not fall into the `undefined === undefined` branch. Do you want to open a PR?

**Reply by matty-at-rdc** [reply]
Sure -- Or at least I'll give it a shot!

**Reply by JLHwung** [reply]
Thanks! Feel free to ask any questions in the slack channel.

**Reply by matty-at-rdc** [reply]
I do have a question @JLHwung and I attempted to click the "Get an Invite to The Slack Channel" link from [CONTRIBUTING.md](https://github.com/babel/babel/blob/main/CONTRIBUTING.md#not-sure-where-to-start), but got an error indicating the link no longer works 😕


**EDIT: I used [babel-slack-invite-link.netlify.app](https://babel-slack-invite-link.netlify.app/) and it works!**

**Reply by JLHwung** [reply]
@matty-at-rdc Yeah we have updated the invite link and waiting for DNS to catch up, tracked in https://github.com/babel/babel/issues/15538.
